draft-ietf-aipref-vocab.md

```diff
@@ -203,11 +203,144 @@ If arrangements, such as contracts exist that explicitly permit the use of that
+an ASCII decoder or UTF-8 decoder {{?UTF8=RFC3629}} can be used.
+This results in the strings that this document uses.
+
+Processing (see {{processing}}) requires a sequence of bytes,
+so any format that uses strings needs to encode strings first.
+Again, this process can use ASCII or UTF-8.
+
+
+## Syntax Extensions
+
+There are two ways by which this syntax might be extended:
+the addition of new labels and the addition of parameters.
+
+New labels might be defined to correspond to new usage categories.
+{{vocab-extension}} addresses the considerations for defining new categories.
+New labels might also be defined for other types of extension
+that do not assign a preference to a usage category.
+In either case, when processing a parsed Dictionary to obtain preferences,
+any unknown labels MUST be ignored.
+
+The Dictionary syntax ({{Section 3.2 of !FIELDS}}) can associate parameters
+with each key-value pair.
+This document does not define any semantics for any parameters that might be included.
+When processing a parsed Dictionary to obtain preferences,
+any unknown parameters MUST be ignored.
+
+
+## Processing Algorithm {#processing}
+
+To process a series of bytes to recover the expressed preferences,
+those bytes are parsed into a Dictionary ({{Section 4.2.2 of !FIELDS}}),
+then preferences are assigned to each usage category in the vocabulary.
+
+The parsing algorithm for a Dictionary
+produces a keyed collection of values,
+each with a possibly-empty set of parameters.
+The parsing process guarantees that each key has at most one value and parameters.
+
+To obtain preferences for each of the categories in the vocabulary,
+iterate through the categories.
+For the label that corresponds to that category (see {{t-category-labels}}),
+obtain the corresponding value from the collection,
+disregarding any parameters.
+A preference is assigned as follows:
+
+* If the value is a Token with a value of "y" (not including quotes),
+  the associated preference is to allow that category of use.
+
+* If the value is a Token with a value of "n" (not including quotes),
+  the associated preference is to disallow that category of use.
+
+* Otherwise, a preference is not expressed for that category of use.
+
+Note that this last alternative includes
+the key being absent from the collection,
+values that are not Tokens,
+and Token values that are other than "y" or "n".
+All of these are not errors,
+they only result in no preference being inferred.
+
+An important note about this process and format is that,
+if the same key appears multiple times,
+only the last value is taken.
+This means that duplicating the same key could result in unexpected outcomes.
+For example, the following expresses no preferences:
+
+~~~
+ai=y, ai="n", genai=n, genai, tdm=n, tdm=()
+~~~
+
+If the parsing of the Dictionary fails, no preferences are expressed.
+This includes where keys include uppercase characters,
+as this format is case sensitive.
```

Comment:
If this format is based on bytes and not strings - there is no such thing as case sensitivity, as that only applies to strings with a text encoding...which this does not have!